app/src/main/java/com/github/ravifrancesco/softwaredevproject/Point.java

```diff
@@ -139,4 +144,171 @@ public double getAltitude() {
+     */
+    public double setVerticalBearing(Point startPoint) {
+        this.verticalBearing = computeVerticalBearing(startPoint);
+        return verticalBearing;
+    }
+
+    /**
+     * Calculates the horizontal angle between the point and another point
+     * Using the great circle path formula
+     * @param startPoint point from where we calculate the bearing
+     * @return Angle in degrees between 0 and 360
+     */
+    private double computeHorizontalBearing(Point startPoint){
+        double startLat = Math.toRadians(startPoint.getLatitude());
+        double startLon = startPoint.getLongitude();
+        double endLat = Math.toRadians(this.getLatitude());
+        double endLon = this.getLongitude();
+
+        double deltaLong = Math.toRadians(endLon - startLon);
+        double y = Math.sin(deltaLong) * Math.cos(Math.toRadians(endLat));
+        double x = Math.cos(startLat) * Math.sin(endLat)
+                - Math.sin(startLat) * Math.cos(endLat) * Math.cos(deltaLong);
+        return (Math.toDegrees(Math.atan2(y, x)) + 360) % 360;
+    }
+
+    /**
+     * Calculates the vertical angle between a startpoint and the POIPoint
+     * See: http://cosinekitty.com/compass.html
+     * @param startPoint Point from where the angle is calculated
+     * @return Angle in degrees between 0° (mountain exactly below) and
+     * 180° (mountain exactly above), horizon is 90°
+     */
+    private double computeVerticalBearing(Point startPoint){
+        Pair<Point, Point> ap = locationToPoint(startPoint);
+        Pair<Point, Point> bp = locationToPoint(this);
+        Point apGeocentric = ap.first;
+        Point apGeodetic = ap.second;
+        Point bpGeocentric = bp.first;
+
+        assert bpGeocentric != null;
+        assert apGeocentric != null;
+        Point bma = normalizeVectorDiff(bpGeocentric, apGeocentric);
+
+        assert apGeodetic != null;
+        double delta = bma.getLatitude() * apGeodetic.getLatitude() +
+                bma.getLongitude() * apGeodetic.getLongitude() + bma.getAltitude() * apGeodetic.getAltitude();
+
+        double elevation = Math.acos(-1*delta);
+        return Math.toDegrees(elevation);
+    }
+
+    /**
+     * Uses two points to create a new normalized Point
+     * @param b first Point
+     * @param a second Point
+     * @return Point that is normalized
+     */
+    private Point normalizeVectorDiff(Point b, Point a){
+        // Calculate norm(b-a), where norm divides a vector by its length to produce a unit vector.
+        double dx = b.getLatitude() - a.getLatitude();
+        double dy = b.getLongitude() - a.getLongitude();
+        double dz = b.getAltitude() - a.getAltitude();
+        double dist2 = dx*dx + dy*dy + dz*dz;
+        double dist = Math.sqrt(dist2);
+        return new Point(dx/dist, dy/dist, dz/dist);
+    }
+
+    /**
+     * Compute the location to the point
+     * @param c Point that gets computed
+     * @return new Point with computed location
```

Comment:
It actually returns one point but with the geocentric and geodetic coordinate system. I use a Pair of points because I find it better than an array of all the values.

Reply:
Updated comments